packages/@o3r/rules-engine/src/engine/structure.ts

```diff
@@ -136,6 +136,14 @@ export interface Ruleset {
   rules: Rule[];
   /** Optional date range where the ruleset will be executed*/
   validityRange?: {from?: string; to?: string};
-  /** Component linked to the ruleset, if set it will disable the ruleset execution per default, waiting to a subscription */
+  /**
+   * Components linked to the ruleset. If present the ruleset will not be active by default.
+   * 'or' condition: If at least one component has subscribed, the ruleset will become active.
+   */
+  linkedComponents?: Record<'or', {library: string; name: string}[]>;
```

Comment:
I know we already had some discussion on the format but what will happened when we will add `and`
People will be able to have an object like this 
```json5
{
  linkedComponents: {
    or: [...],
    and: [...]
}
```
And I don't see how we can handle `or` and `and` at the same time
I think we should have something like this
```typescript
type ComponentList = {library: string; name: string}[];
type GroupOperator = 'or';
type GroupComponents = 
   | { operator: GroupOperator, components: ComponentList } 
   | { operator: GroupOperator, groups: GroupComponents[] }
linkedComponents?: GroupComponents;
```
So like this we will be able to support use case like this
```json5
linkedComponents: {
  operator: 'or',
  groups: [
    {
      operator: 'and',
      components: [
        {library: '@o3r/lib'; name: 'AndExample1Component'}, 
        {library: '@o3r/lib'; name: 'AndExample2Component'}
      ]
    },
    {
       operator: 'or',
       components: [
         {library: '@o3r/lib'; name: 'OrExample1Component'}, 
         {library: '@o3r/lib'; name: 'OrExample2Component'}
       ]
    }
  ]
}
```
and would lead to this logic
`(AndExample1Component && AndExample2Component) || (OrExample1Component || OrExample2Component)`

Reply:
I think we are going to far here with the nested groups.
We don't even have the usecase of a simple 'AND'.
With the actual implementation we support a list of linked components driven by 'OR' operator and if we want we'll have another list with 'AND' operator. 
If in some day we'll want to support the 'AND' we will need one selector for 'OR' components and one selector for 'AND'. Nowadays the 'OR' selector is in place (the default one we have).
I let the others give their opinion too :)